timely_beliefs/beliefs/classes.py

```diff
@@ -351,36 +379,219 @@ def knowledge_horizons(self) -> pd.TimedeltaIndex:
+    def belief_history(
+        self,
+        event_start: datetime,
+        belief_time_window: Tuple[Optional[datetime], Optional[datetime]] = (
+            None,
+            None,
+        ),
+        belief_horizon_window: Tuple[Optional[timedelta], Optional[timedelta]] = (
+            None,
+            None,
+        ),
+    ) -> "BeliefsDataFrame":
+        """Select all beliefs about a single event, identified by the event's start time.
+        Optionally select a history of beliefs formed within a certain time window.
+        Alternatively, select a history of beliefs formed a certain horizon window before knowledge time (with negative
+        horizons indicating post knowledge time).
+
+        :Example:
+
+        >>> # Window selecting beliefs formed before June 20th 2018
+        >>> df.belief_history(event_start, belief_time_window=(None, datetime(2018, 6, 20)))
+        >>> # Window selecting beliefs formed from 5 to 10 hours before knowledge time
+        >>> df.belief_history(event_start, belief_horizon_window=(timedelta(hours=5), timedelta(hours=10)))
+        >>> # Window selecting beliefs formed from 2 hours after to 10 hours before knowledge time
+        >>> df.belief_history(event_start, belief_horizon_window=(timedelta(hours=-2), timedelta(hours=10)))
+
+        :param: event_start: start time of the event
+        :param: belief_time_window: optional tuple specifying a time window within which beliefs should have been formed
+        :param: belief_horizon_window: optional tuple specifying a horizon window (e.g. between 1 and 2 hours before the event value could have been known)
+        """
+        df = self.xs(
+            enforce_utc(event_start), level="event_start", drop_level=False
+        ).sort_index()
+        if belief_time_window[0] is not None:
+            df = df[df.index.get_level_values("belief_time") >= belief_time_window[0]]
+        if belief_time_window[1] is not None:
+            df = df[df.index.get_level_values("belief_time") <= belief_time_window[1]]
+        if belief_horizon_window != (None, None):
+            if belief_time_window != (None, None):
+                raise ValueError(
+                    "Cannot pass both a belief time window and belief horizon window."
+                )
+            df = df.convert_index_from_belief_time_to_horizon()
+            if belief_horizon_window[0] is not None:
+                df = df[
+                    df.index.get_level_values("belief_horizon")
+                    >= belief_horizon_window[0]
+                ]
+            if belief_horizon_window[1] is not None:
+                df = df[
+                    df.index.get_level_values("belief_horizon")
+                    <= belief_horizon_window[1]
+                ]
+            df = df.convert_index_from_belief_horizon_to_time()
+        df.index = df.index.droplevel("event_start")
+        return df
+
+    @hybrid_method
+    def fixed_horizon(
```

Comment:
I always have a mental itch when talking about the concepts of "fixed horizon" and "rolling horizon". Because the rolling horizon implies a constant horizon for each different event and the fixed horizon implies a changing horizon for each different event.

I agree that a name change is in order, but I think we can do better than changing `fixed_horizon` to `fixed_beliefs`, since the act of fixing something should pertain to timing more explicitly. Perhaps ` fixed_viewpoint`? And likewise, ` rolling_viewpoint` for `rolling_horizon`. So we'd get:

`fixed_viewpoint` -> the horizon increases as you look further ahead.
`rolling_viewpoint` -> the horizon stays the same, because your viewpoint moves with you as you look further ahead.